### PR TITLE
FIX bookmark URLs get truncated to 255 chars

### PR DESCRIPTION
**Pull request by atm-florianm**
# Fix bookmark URLs can’t be longer than 255c
Currently, bookmark URLs are stored as a `VARCHAR(255)`. Some URLs are longer than this and get silently truncated (which could be dangerous in some rare cases if the truncated URL is still valid but points to a different object, especially if the user doesn’t notice it).

The fix changes the type to `TEXT`. In MySQL, `TEXT` fields can’t have a `UNIQUE` constraint, so I removed it too. I also removed the `NOT NULL` option : apparently, PgSQL can’t have `NOT NULL` without a default value, but MySQL can’t have a default value for `TEXT`.

### CODE REVIEW

**Comment by frederic34** [comment]
https://github.com/Dolibarr/dolibarr/pull/11839

**Comment by eldy** [comment]
As it changes structure, it would be merged for v12.
Can you resubmit on v12 (the migration file must also be v11->v12) ?

**Comment by atm-florianm** [comment]
> As it changes structure, it would be merged for v12.
> Can you resubmit on v12 (the migration file must also be v11->v12) ?

I modified the PR accordingly by moving the 3 migration lines to a new `11.0.0-12.0.0.sql` file. I hope this is OK.

File: htdocs/install/mysql/tables/llx_bookmark.sql [hunk 1]
```diff
@@ -21,7 +21,7 @@ create table llx_bookmark
   rowid       integer AUTO_INCREMENT PRIMARY KEY,
   fk_user     integer NOT NULL,
   dateb       datetime,
-  url         varchar(255) NOT NULL,
+  url         TEXT,
```
Comment (eldy) [inline htdocs/install/mysql/tables/llx_bookmark.sql]: Can't you keep the Not null ?

Comment (atm-florianm) [reply htdocs/install/mysql/tables/llx_bookmark.sql]: I don't know how to do it while keeping PgSQL compatibility. Postgres doesn’t allow NOT NULL without a default value (at least when altering a table), but MySQL doesn’t allow default values for TEXT.

**Pull request merged**